source/MaterialXTest/MaterialXCore/Document.cpp

```diff
@@ -116,3 +119,215 @@ TEST_CASE("Document", "[document]")
     // Validate the combined document.
     REQUIRE(doc->validate());
 }
+
+void printDifferences(mx::ElementEquivalenceResult& result, const std::string& label)
```

Comment:
Rather than baking in the use of `std::cout` in this helper method, it might be better to follow the example of `prettyPrint` and return the results as a standard string, allowing the user to send them to any logging system that they prefer:

https://github.com/AcademySoftwareFoundation/MaterialX/blob/main/source/MaterialXCore/Element.cpp#L666